pages/signup.tsx

```
@@ -127,26 +134,63 @@ const SignupForm: React.FC<SignupFormProps> = ({
   )
 }
 
-const Signup: React.FC = () => {
+const SignUpPage: React.FC = () => {
   const [signupSuccess, setSignupSuccess] = useState(false)
-  const [signupErrors, setSignupErrors] = useState({} as SignupErrors)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [signupErrors, setSignupErrors] = useState<string[]>([])
+  const [signupUser] = useMutation(SIGNUP_USER)
   const handleSubmit = async (values: Values) => {
-    const data = await signupUser(values)
-    if (data.success) {
-      setSignupSuccess(true)
-    } else {
-      setSignupErrors(Object.assign({}, data.errorMessage))
+    setIsSubmitting(true)
+    try {
+      const { data } = await signupUser({ variables: values })
+      if (data.signup.success) {
+        return setSignupSuccess(true)
+      }
+      const err = new Error(
+        'Server cannot be reached. Please try again. If this problem persists, please send an email to support@c0d3.com'
+      )
+      throw err
```

You can just throw the error since the err isn't being used for anything.